pkg/appgw/certificates.go

```diff
@@ -32,20 +32,11 @@ func (c *appGwConfigBuilder) getSslCertificates(cbCtx *ConfigBuilderContext) *[]
 
 	var sslCertificates []n.ApplicationGatewaySslCertificate
 	for secretID, cert := range secretIDCertificateMap {
-		sslCertificates = append(sslCertificates, newCert(secretID, cert))
+		sslCertificates = append(sslCertificates, c.newCert(secretID, cert))
 	}
 
 	if cbCtx.EnableBrownfieldDeployment {
-		er := brownfield.NewExistingResources(c.appGw, cbCtx.ProhibitedTargets, nil)
-
-		// Certs we obtained from App Gateway - we segment them into ones AGIC is and is not allowed to change.
-		existingBlacklisted, existingNonBlacklisted := er.GetBlacklistedCertificates()
-
-		brownfield.LogCertificates(existingBlacklisted, existingNonBlacklisted, sslCertificates)
-
-		// MergePools would produce unique list of pools based on Name. Blacklisted pools, which have the same name
-		// as a managed pool would be overwritten.
-		sslCertificates = brownfield.MergeCerts(existingBlacklisted, sslCertificates)
+		sslCertificates = brownfield.MergeCerts(*c.appGw.SslCertificates, sslCertificates)
```

Comment:
We should leave a comment here documenting the choices we deliberately made... (to not garbage-collect certs etc.)